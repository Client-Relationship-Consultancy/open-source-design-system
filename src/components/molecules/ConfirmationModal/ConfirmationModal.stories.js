import React from "react"
import { storiesOf } from "@storybook/react"
import styled from "styled-components"
import { text } from "@storybook/addon-knobs"

import { ConfirmationModal } from "../../../index"



storiesOf("Modals", module)
  .add("Confirmation Modal without Title", () => (
    <ConfirmationModal
      isOpen
      yesButtonLabel={text("Yes button label", "Button")}
      noButtonLabel={text("No button label", "Cancel")}
    >
      Without a title, we can have text here also.
      <br />
      <br />
      You have clicked on another user whilst still in progress of editing the permissions of
      Bethany B. If you change users now you will lose the current changes you have made.
      <br />
      <br />
      Are you sure want to switch to another user?
    </ConfirmationModal>
  ))
  .add("Confirmation Modal with Title", () => (
    <ConfirmationModal
      isOpen
      yesButtonLabel={text("Yes button label", "Button")}
      noButtonLabel={text("No button label", "Cancel")}
      title="Title of the modal"
    >
      You have clicked on another user whilst still in progress of editing the permissions of
      Bethany B. If you change users now you will lose the current changes you have made.
      <br />
      <br />
      <b>Are you sure want to switch to another user?</b>
    </ConfirmationModal>
  ))
