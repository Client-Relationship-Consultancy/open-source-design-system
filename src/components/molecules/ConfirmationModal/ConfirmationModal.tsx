import React from "react";
import styled from "styled-components";
import ReactModal from "react-modal";
import Button, { IconButton } from "../../atoms/Button";
import zIndexes from "../../../zIndexes";
import { colourPalette } from "../../../brandColours";

interface IWithTitleProps extends React.HTMLProps<HTMLDivElement> {
  withTitle: boolean;
}

export const HeaderBar = styled.div<IWithTitleProps>`
  width: 100%;
  height: ${({ withTitle }) => (withTitle ? "3.5rem" : "2.875rem")};
  border-bottom: 0.0625rem solid
    ${({ withTitle, theme }) => (withTitle ? theme.primary.main.hex : "transparent")};
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  justify-content: space-between;
  padding: 0 0.6rem;
  h1 {
    font-size: 1.375rem;
    font-weight: 500;
    line-height: 1.65rem;
    color: ${({ theme }) => theme.primary.main.hex};
    margin-left: 1.4rem;
  }
`;
HeaderBar.defaultProps = {
  theme: colourPalette.examplePalette,
};

const InsideContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

const ButtonRow = styled.div`
  display: flex;
  flex-direction: row;
  margin-bottom: 1rem;
  justify-content: flex-end;
  padding: 0 1.125rem;
  > * + * {
    margin-left: 1.125rem;
  }
`;

export const ContentContainer = styled.p<IWithTitleProps>`
  display: flex;
  justify-content: flex-start;
  flex-direction: column;
  padding: 0 2rem;
  font-size: 1rem;
  font-weight: 300;
  line-height: 1.2rem;
  font-style: normal;
  margin: 1.3125rem 0;
  margin-top: ${({ withTitle }) => (withTitle ? "1.3125rem" : "0")};
`;


const customStyles = {
  overlay: {
    zIndex: zIndexes.modal,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
  content: {
    overflow:"visible",
    margin: "auto",
    top: "auto",
    right: "auto",
    bottom: "auto",
    left: "auto",
    padding: 0,
    width: "43.75rem",
    borderRadius: "0.625rem",
    boxShadow: "0px 0px 16px #696969",
  },
};

export interface IConfirmationModal {
  isOpen: boolean;
  onYes: (e: React.SyntheticEvent) => void;
  onNo: (e: React.SyntheticEvent) => void;
  id?: string;
  className?: string;
  classNameHooks?: {
    container?: string;
  };
  yesButtonLabel?: string | React.ReactNode;
  noButtonLabel?: string | React.ReactNode;
  title?: string | React.ReactNode;
}
export const ConfirmationModal: React.FunctionComponent<IConfirmationModal> = (props) => {
  const {
    isOpen,
    id,
    className,
    children,
    onYes,
    onNo,
    classNameHooks,
    yesButtonLabel = "Yes",
    noButtonLabel = "No",
    title,
  } = props;
  return (
    <ReactModal isOpen={isOpen} style={customStyles} id={id} className={className}>
      <InsideContainer className={classNameHooks?.container}>
        <HeaderBar withTitle={!!title}>
          <IconButton buttonSize="large" buttonType="ghost" icon="times" onClick={onNo} />
          {title && <h1>{title}</h1>}
        </HeaderBar>
        <ContentContainer withTitle={!!title}>{children}</ContentContainer>
        <ButtonRow>
          <Button onClick={onNo} buttonType="ghost" buttonSize="medium">
            <span className="modal_noButtonLabel">{noButtonLabel}</span>
          </Button>
          <Button onClick={onYes} buttonType="primary" buttonSize="medium">
            <span className="modal_yesButtonLabel">{yesButtonLabel}</span>
          </Button>
        </ButtonRow>
      </InsideContainer>
    </ReactModal>
  );
};

ConfirmationModal.displayName = "ConfirmationModal";

export default ConfirmationModal;
