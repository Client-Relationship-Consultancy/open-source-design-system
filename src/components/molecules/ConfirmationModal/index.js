export { ConfirmationModal, IConfirmationModal } from "./ConfirmationModal"
