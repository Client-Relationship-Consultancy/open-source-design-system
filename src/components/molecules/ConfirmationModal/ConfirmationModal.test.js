import React from "react"
import { shallow, mount } from "enzyme"
import ConfirmationModal, { HeaderBar, ContentContainer } from "./ConfirmationModal"
import "jest-styled-components"

describe("ConfirmationModal Component Test", () => {
  it("Match last snapshot", () => {
    const component = shallow(
      <ConfirmationModal isOpen>I am a lovely confirmation modal</ConfirmationModal>,
    )
    expect(component).toMatchSnapshot()
  })

  it("Match last snapshot with title", () => {
    const component = shallow(
      <ConfirmationModal isOpen title="test title">I am a lovely confirmation modal</ConfirmationModal>,
    )
    expect(component).toMatchSnapshot()
  })

  it("check function in styled components", () => {
    const headerBar = mount(<HeaderBar />)
    expect(headerBar).toMatchSnapshot()
  })

  it("check function in styled components with title", () => {
    const headerBar = mount(<HeaderBar withTitle />)
    expect(headerBar).toMatchSnapshot()
  })

  it("check function in styled components ContentContainer without title", () => {
    const contentContainer = mount(<ContentContainer />)
    expect(contentContainer).toMatchSnapshot()
  })

  it("check function in styled components ContentContainer with title", () => {
    const contentContainer = mount(<ContentContainer withTitle />)
    expect(contentContainer).toMatchSnapshot()
  })

  it("should show 'Yes' and 'No' when no custom button label is passed", () => {
    const component = shallow(
      <ConfirmationModal isOpen>I am a lovely confirmation modal</ConfirmationModal>,
    )

    expect(component.find("ButtonText.yesButton").text()).toEqual("Yes")
    expect(component.find("ButtonText.noButton").text()).toEqual("No")
  })

  it("should accept a string/JSX message for the yes button", () => {
    const component = shallow(
      <ConfirmationModal isOpen yesButtonLabel="I like to confirm">
        I am a lovely confirmation modal
      </ConfirmationModal>,
    )

    expect(component.find("ButtonText.yesButton").text()).toEqual("I like to confirm")

    const YesJSX = () => <div>Confirm</div>
    component.setProps({ yesButtonLabel: <YesJSX /> })
    expect(component.find("ButtonText.yesButton").contains(<YesJSX />)).toEqual(true)
  })

  it("should accept a string/JSX message for the no button", () => {
    const component = shallow(
      <ConfirmationModal isOpen noButtonLabel="Cancel">
        I am a lovely confirmation modal
      </ConfirmationModal>,
    )

    expect(component.find("ButtonText.noButton").text()).toEqual("Cancel")

    const NoJSX = () => <div>Nope</div>
    component.setProps({ noButtonLabel: <NoJSX /> })
    expect(component.find("ButtonText.noButton").contains(<NoJSX />)).toEqual(true)
  })
})
