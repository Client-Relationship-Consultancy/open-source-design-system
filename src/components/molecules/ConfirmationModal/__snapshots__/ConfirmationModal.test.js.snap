// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal Component Test Match last snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentElement={[Function]}
  isOpen={true}
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "bottom": 0,
        "height": "300px",
        "left": 0,
        "margin": "auto",
        "overflow": "visible",
        "padding": 0,
        "right": 0,
        "top": 0,
        "width": "700px",
      },
      "overlay": Object {
        "zIndex": 100,
      },
    }
  }
>
  <ConfirmationModal__InsideContainer>
    <ConfirmationModal__HeaderBar
      theme={
        Object {
          "action": Object {
            "dark": Object {
              "RGB": "79,38,98",
              "hex": "#4E2662",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "173,101,207",
              "hex": "#AD65CF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "116,60,143",
              "hex": "#743C8F",
              "on": "#FFFFFF",
            },
          },
          "background": Object {
            "RGB": "246,246,246",
            "hex": "#F6F6F6",
            "on": "#272727",
          },
          "black": Object {
            "main": Object {
              "RGB": "39,39,39",
              "hex": "#272727",
              "on": "#FFFFFF",
            },
            "tint20": Object {
              "RGB": "236,236,236",
              "hex": "#ECECEC",
              "on": "#272727",
            },
            "tint40": Object {
              "RGB": "199,199,199",
              "hex": "#C7C7C7",
              "on": "#272727",
            },
            "tint60": Object {
              "RGB": "160,160,160",
              "hex": "#A0A0A0",
              "on": "#FFFFFF",
            },
            "tint80": Object {
              "RGB": "105,105,105",
              "hex": "#696969",
              "on": "#FFFFFF",
            },
          },
          "complimentary": Object {
            "dark": Object {
              "RGB": "206,185,0",
              "hex": "#CEB900",
              "on": "#272727",
            },
            "light": Object {
              "RGB": "255,245,157",
              "hex": "#FFF59D",
              "on": "#272727",
            },
            "main": Object {
              "RGB": "255,238,82",
              "hex": "#FFEE52",
              "on": "#272727",
            },
          },
          "error": Object {
            "dark": Object {
              "RGB": "197,73,1",
              "hex": "#C54901",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "255,115,32",
              "hex": "#FF7320",
              "on": "#FFFFFF",
            },
          },
          "primary": Object {
            "dark": Object {
              "RGB": "102,22,20",
              "hex": "#661614",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "218,52,47",
              "hex": "#DA342F",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "164,40,36",
              "hex": "#A42824",
              "on": "#FFFFFF",
            },
          },
          "secondary": Object {
            "dark": Object {
              "RGB": "5,38,59",
              "hex": "#05263B",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "89,156,175",
              "hex": "#599CAF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "11,70,107",
              "hex": "#0B466B",
              "on": "#FFFFFF",
            },
          },
          "surface": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
          "white": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
        }
      }
    />
    <ConfirmationModal__ContentContainer>
      I am a lovely confirmation modal
    </ConfirmationModal__ContentContainer>
    <ConfirmationModal__ButtonRow>
      <Button
        buttonType="primary"
      >
        <ButtonText
          className="yesButton"
        >
          Yes
        </ButtonText>
      </Button>
      <Button
        buttonType="primaryOutline"
      >
        <ButtonText
          className="noButton"
        >
          No
        </ButtonText>
      </Button>
    </ConfirmationModal__ButtonRow>
  </ConfirmationModal__InsideContainer>
</Modal>
`;

exports[`ConfirmationModal Component Test check function in styled components 1`] = `
.c0 {
  width: 100%;
  background-color: #A42824;
  height: 40px;
}

<ConfirmationModal__HeaderBar
  theme={
    Object {
      "action": Object {
        "dark": Object {
          "RGB": "79,38,98",
          "hex": "#4E2662",
          "on": "#FFFFFF",
        },
        "light": Object {
          "RGB": "173,101,207",
          "hex": "#AD65CF",
          "on": "#FFFFFF",
        },
        "main": Object {
          "RGB": "116,60,143",
          "hex": "#743C8F",
          "on": "#FFFFFF",
        },
      },
      "background": Object {
        "RGB": "246,246,246",
        "hex": "#F6F6F6",
        "on": "#272727",
      },
      "black": Object {
        "main": Object {
          "RGB": "39,39,39",
          "hex": "#272727",
          "on": "#FFFFFF",
        },
        "tint20": Object {
          "RGB": "236,236,236",
          "hex": "#ECECEC",
          "on": "#272727",
        },
        "tint40": Object {
          "RGB": "199,199,199",
          "hex": "#C7C7C7",
          "on": "#272727",
        },
        "tint60": Object {
          "RGB": "160,160,160",
          "hex": "#A0A0A0",
          "on": "#FFFFFF",
        },
        "tint80": Object {
          "RGB": "105,105,105",
          "hex": "#696969",
          "on": "#FFFFFF",
        },
      },
      "complimentary": Object {
        "dark": Object {
          "RGB": "206,185,0",
          "hex": "#CEB900",
          "on": "#272727",
        },
        "light": Object {
          "RGB": "255,245,157",
          "hex": "#FFF59D",
          "on": "#272727",
        },
        "main": Object {
          "RGB": "255,238,82",
          "hex": "#FFEE52",
          "on": "#272727",
        },
      },
      "error": Object {
        "dark": Object {
          "RGB": "197,73,1",
          "hex": "#C54901",
          "on": "#FFFFFF",
        },
        "main": Object {
          "RGB": "255,115,32",
          "hex": "#FF7320",
          "on": "#FFFFFF",
        },
      },
      "primary": Object {
        "dark": Object {
          "RGB": "102,22,20",
          "hex": "#661614",
          "on": "#FFFFFF",
        },
        "light": Object {
          "RGB": "218,52,47",
          "hex": "#DA342F",
          "on": "#FFFFFF",
        },
        "main": Object {
          "RGB": "164,40,36",
          "hex": "#A42824",
          "on": "#FFFFFF",
        },
      },
      "secondary": Object {
        "dark": Object {
          "RGB": "5,38,59",
          "hex": "#05263B",
          "on": "#FFFFFF",
        },
        "light": Object {
          "RGB": "89,156,175",
          "hex": "#599CAF",
          "on": "#FFFFFF",
        },
        "main": Object {
          "RGB": "11,70,107",
          "hex": "#0B466B",
          "on": "#FFFFFF",
        },
      },
      "surface": Object {
        "RGB": "255,255,255",
        "hex": "#FFFFFF",
        "on": "#272727",
      },
      "white": Object {
        "RGB": "255,255,255",
        "hex": "#FFFFFF",
        "on": "#272727",
      },
    }
  }
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "_foldedDefaultProps": Object {
          "theme": Object {
            "action": Object {
              "dark": Object {
                "RGB": "79,38,98",
                "hex": "#4E2662",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "173,101,207",
                "hex": "#AD65CF",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "116,60,143",
                "hex": "#743C8F",
                "on": "#FFFFFF",
              },
            },
            "background": Object {
              "RGB": "246,246,246",
              "hex": "#F6F6F6",
              "on": "#272727",
            },
            "black": Object {
              "main": Object {
                "RGB": "39,39,39",
                "hex": "#272727",
                "on": "#FFFFFF",
              },
              "tint20": Object {
                "RGB": "236,236,236",
                "hex": "#ECECEC",
                "on": "#272727",
              },
              "tint40": Object {
                "RGB": "199,199,199",
                "hex": "#C7C7C7",
                "on": "#272727",
              },
              "tint60": Object {
                "RGB": "160,160,160",
                "hex": "#A0A0A0",
                "on": "#FFFFFF",
              },
              "tint80": Object {
                "RGB": "105,105,105",
                "hex": "#696969",
                "on": "#FFFFFF",
              },
            },
            "complimentary": Object {
              "dark": Object {
                "RGB": "206,185,0",
                "hex": "#CEB900",
                "on": "#272727",
              },
              "light": Object {
                "RGB": "255,245,157",
                "hex": "#FFF59D",
                "on": "#272727",
              },
              "main": Object {
                "RGB": "255,238,82",
                "hex": "#FFEE52",
                "on": "#272727",
              },
            },
            "error": Object {
              "dark": Object {
                "RGB": "197,73,1",
                "hex": "#C54901",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "255,115,32",
                "hex": "#FF7320",
                "on": "#FFFFFF",
              },
            },
            "primary": Object {
              "dark": Object {
                "RGB": "102,22,20",
                "hex": "#661614",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "218,52,47",
                "hex": "#DA342F",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "164,40,36",
                "hex": "#A42824",
                "on": "#FFFFFF",
              },
            },
            "secondary": Object {
              "dark": Object {
                "RGB": "5,38,59",
                "hex": "#05263B",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "89,156,175",
                "hex": "#599CAF",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "11,70,107",
                "hex": "#0B466B",
                "on": "#FFFFFF",
              },
            },
            "surface": Object {
              "RGB": "255,255,255",
              "hex": "#FFFFFF",
              "on": "#272727",
            },
            "white": Object {
              "RGB": "255,255,255",
              "hex": "#FFFFFF",
              "on": "#272727",
            },
          },
        },
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "ConfirmationModal__HeaderBar-sc-1lf36yh-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "width:100%;background-color:",
            [Function],
            ";height:40px;",
          ],
        },
        "displayName": "ConfirmationModal__HeaderBar",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "ConfirmationModal__HeaderBar-sc-1lf36yh-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "action": Object {
          "dark": Object {
            "RGB": "79,38,98",
            "hex": "#4E2662",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "173,101,207",
            "hex": "#AD65CF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "116,60,143",
            "hex": "#743C8F",
            "on": "#FFFFFF",
          },
        },
        "background": Object {
          "RGB": "246,246,246",
          "hex": "#F6F6F6",
          "on": "#272727",
        },
        "black": Object {
          "main": Object {
            "RGB": "39,39,39",
            "hex": "#272727",
            "on": "#FFFFFF",
          },
          "tint20": Object {
            "RGB": "236,236,236",
            "hex": "#ECECEC",
            "on": "#272727",
          },
          "tint40": Object {
            "RGB": "199,199,199",
            "hex": "#C7C7C7",
            "on": "#272727",
          },
          "tint60": Object {
            "RGB": "160,160,160",
            "hex": "#A0A0A0",
            "on": "#FFFFFF",
          },
          "tint80": Object {
            "RGB": "105,105,105",
            "hex": "#696969",
            "on": "#FFFFFF",
          },
        },
        "complimentary": Object {
          "dark": Object {
            "RGB": "206,185,0",
            "hex": "#CEB900",
            "on": "#272727",
          },
          "light": Object {
            "RGB": "255,245,157",
            "hex": "#FFF59D",
            "on": "#272727",
          },
          "main": Object {
            "RGB": "255,238,82",
            "hex": "#FFEE52",
            "on": "#272727",
          },
        },
        "error": Object {
          "dark": Object {
            "RGB": "197,73,1",
            "hex": "#C54901",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "255,115,32",
            "hex": "#FF7320",
            "on": "#FFFFFF",
          },
        },
        "primary": Object {
          "dark": Object {
            "RGB": "102,22,20",
            "hex": "#661614",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "218,52,47",
            "hex": "#DA342F",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "164,40,36",
            "hex": "#A42824",
            "on": "#FFFFFF",
          },
        },
        "secondary": Object {
          "dark": Object {
            "RGB": "5,38,59",
            "hex": "#05263B",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "89,156,175",
            "hex": "#599CAF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "11,70,107",
            "hex": "#0B466B",
            "on": "#FFFFFF",
          },
        },
        "surface": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
        "white": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
      }
    }
  >
    <div
      className="c0"
    />
  </StyledComponent>
</ConfirmationModal__HeaderBar>
`;
